### PR TITLE
Fix polling logic by returning error as part of return statement

### PR DESCRIPTION
**Pull request by abhijeet-dhumal**
## Description

Fixed ODH e2e tests polling logic by returning error as part of return statement

## How Has This Been Tested?

Manually

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by sutaakar** [comment]
Can you please also fix the code:
```
	test.Eventually(func() []RayJobDetailsResponse {
		rayJobs, err := rayClient.GetJobs()
		test.Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to fetch ray-jobs : %s", err))
		return *rayJobs
	}, TestTimeoutMedium, 1*time.Second).Should(HaveLen(1), "Ray job not found")
```

File: tests/odh/mnist_ray_test.go [hunk 1]
```diff
@@ -141,10 +141,13 @@ func mnistRay(t *testing.T, numGpus int) {
 	test.Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to create new raycluster client: %s", err))
 
 	// wait until rayjob exists
-	test.Eventually(func() []RayJobDetailsResponse {
+	test.Eventually(func() ([]RayJobDetailsResponse, error) {
 		rayJobs, err := rayClient.GetJobs()
 		test.Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to fetch ray-jobs : %s", err))
```
Comment (sutaakar) [inline tests/odh/mnist_ray_test.go:146]: Can you remove this line? It is causing the issue.

File: tests/odh/mnist_raytune_hpo_test.go [hunk 2]
```diff
@@ -141,11 +141,14 @@ func mnistRayTuneHpo(t *testing.T, numGpus int) {
 	rayClient, err := NewRayClusterClient(rayClusterClientConfig, test.Config().BearerToken)
 	test.Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to create new raycluster client: %s", err))
 
-	// Wait until the rayjob is created and running
-	test.Eventually(func() []RayJobDetailsResponse {
+	// wait until rayjob exists
+	test.Eventually(func() ([]RayJobDetailsResponse, error) {
 		rayJobs, err := rayClient.GetJobs()
 		test.Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to fetch ray-jobs : %s", err))
```
Comment (sutaakar) [inline tests/odh/mnist_raytune_hpo_test.go:147]: Same here

File: tests/odh/ray_finetune_llm_deepspeed_test.go [hunk 3]
```diff
@@ -129,10 +129,13 @@ func rayFinetuneLlmDeepspeed(t *testing.T, numGpus int, modelName string, modelC
 	test.Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to create new raycluster client: %s", err))
 
 	// wait until rayjob exists
-	test.Eventually(func() []RayJobDetailsResponse {
+	test.Eventually(func() ([]RayJobDetailsResponse, error) {
 		rayJobs, err := rayClient.GetJobs()
 		test.Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to fetch ray-jobs : %s", err))
```
Comment (sutaakar) [inline tests/odh/ray_finetune_llm_deepspeed_test.go:134]: And here

File: tests/odh/mnist_ray_test.go (hunk 1 above)
Comment (abhijeet-dhumal) [reply tests/odh/mnist_ray_test.go]: Ohh, Apologies I remember I removed it but not sure how it got restored :eyes: 
Removed it :checkered_flag:

**Review by sutaakar — APPROVED** [review]

**Pull request merged**